sct-commons/src/main/java/org/lfenergy/compas/sct/commons/api/ControlBlockEditor.java

```diff
@@ -65,4 +66,22 @@ public interface ControlBlockEditor {
      */
     List<SclReportItem> configureNetworkForAllControlBlocks(SCL scd, CBCom cbCom);
 
+    /**
+     * Configure the network for all the ControlBlocks, reusing APPID and MAC-Address from given Communication section when they already exists.
+     * Create (or update if already existing) these elements
+     * - the Communication/SubNetwork/ConnectedAP/GSE element, for the GSEControl blocks
+     * - the Communication/SubNetwork/ConnectedAP/SMV element, for the SampledValueControl blocks
+     * For a ControlBlock (IED.name, LD.inst, GSEControl/SampledValueControl.name), if there is a matching network configuration in communicationToReuse (ConnectedAP.iedName, GSE/SMV.ldInst, GSE/SMV.cbName),
+     * then reuse APPID and MAC-Address from communicationToReuse,
+     * else use range provided in cbCom.
+     * APPID and MAC-Address of communicationToReuse for ControlBlocks that exists in scd are excluded from the ranges given in cbCom.
+     *
+     * @param scd                  input SCD object. The object will be modified with the new GSE and SMV elements
+     * @param cbCom                communication settings to configure Control Block Communication
+     * @param communicationToReuse communication section to search for existing APPID and MAC-Address for ControlBlock
+     * @return list of encountered errors
+     * @see Utils#macAddressToLong(String) for the expected MAC address format
+     */
+    List<SclReportItem> configureNetworkForAllControlBlocks(SCL scd, CBCom cbCom, TCommunication communicationToReuse);
```

Comment:
Given that we simply require the Subnetwork list,why using  `TCommunication`  ? also `SCL` can be used as type

```suggestion
    List<SclReportItem> configureNetworkForAllControlBlocks(SCL scd, CBCom cbCom, SCL targetSCD);
```

Reply:
OK, I change to List of Subnetworks